### ISSUE TITLE
Add section around why we can only support all Frameworks or all Libraries

### ISSUE THREAD

**Issue by orta** [CocoaPods/guides.cocoapods.org#53]
People will be confused. They will presume this is a choice we made for them, but without having the reasoning in the open then no-one knows why. I don't know why. This in part could go along with  https://github.com/CocoaPods/CocoaPods/issues/3029

**Comment by segiddins** [comment]
@mrackwitz / @neonichu could either of you please do this?

**Comment by neonichu** [comment]
I have to pass, because I am also not sure anymore about the reasons.

**Comment by AliSoftware** [comment]
According to @mrackwitz  in Slack just now:

> [It was] to avoid complexity, because there are situations, where you wouldn't be able to ensure that you can build it in that way, and we don't have the tooling in place to check if e.g. a vendored framework is static or dynamic

**Comment by AliSoftware** [comment]
I think we should also explain somewhere in the guides why a pod containing Swift MMUST be compiled as a framework and can't be a static library. The reason being because of ABI compatibility issues, as the runtime is evolving (this can simply be [a simple link to this Apple's blog post](https://developer.apple.com/swift/blog/?id=2) but it needs to be mentioned somewhere in CP guides anyway IMHO)

**Comment by drewcrawford** [comment]
So essentially the [rant](http://sealedabstract.com/rants/on-ios-packaging/) that originally inspired this issue's filing was correct on the facts.  It's not impossible to support both, it's just a question of effort.  

> I think this isn’t done for a few reasons. One, it’s hard. It’s harder to write a tool that can [mix frameworks/libraries/etc.] then it is to do one.... This is all fair. The thing is, either it’s worth doing it to share code with each other or it’s not.

**Comment by scionwest** [comment]
Does this mean that you can not use a library like SSKeychain/AFNetworking with a framework like Alamofire on the same project via cocoapods?

**Comment by segiddins** [comment]
@scionwest no, since those libraries would be built as Frameworks :)

**Comment by scionwest** [comment]
Hmm, it's weird then why as soon as I add use_frameworks! the SSKeychain & LiveSDK (only two non-frameworks I have) can no longer have their types found. I get an 'identifier not found' compiler error. While AlamoFire works fire due to being a Swift framework. I remove yse_frameworks! and AlamoFire can't be found but LiveSDK & SSKeychain can be.

**Comment by orta** [comment]
Probably worth making a stack overflow issue for this. As alamofire is a swift library it has to have `use_frameworks`.

**Comment by scionwest** [comment]
I actually do already have an issue on Stackoverflow for this. http://stackoverflow.com/q/31103831/1815321

**Comment by segiddins** [comment]
You should be using modular `import` statements with frameworks, that will probably solve your issue.

**Comment by neonichu** [comment]
Yep, only module imports work in the bridging header when using frameworks. Also, you might want to get rid of the bridging header entirely, as it is unnecessary when using frameworks, they can be imported directly in Swift.

**Comment by scionwest** [comment]
Alright, I tried removing the bridge header file and reference in my build settings. I then added `import SSKeychain` but the compiler gives me a `No such modeul 'SSKeychain'` error. Is there something special I'm supposed to be doing?
